output/influxdb/output.go

```diff
@@ -196,11 +203,46 @@ func (o *Output) Stop() error {
 	o.logger.Debug("Stopping...")
 	defer o.logger.Debug("Stopped!")
 	o.periodicFlusher.Stop()
+	o.Client.Close()
+	return nil
+}
+
+// createBucket creates the configured bucket if it doesn't exist
+func (o *Output) createBucket() error {
+	ctx := context.Background()
+
+	org, err := o.Client.OrganizationsAPI().FindOrganizationByName(ctx, o.Config.Organization.String)
+	if err != nil {
+		return err
+	}
+
+	buckets := o.Client.BucketsAPI()
+	_, err = buckets.FindBucketByName(ctx, o.Config.Bucket.String)
+	if err == nil {
+		// the bucket already exists
+		return nil
+	}
+
+	// TODO: It has the bucket name in the error message. Can we do a better check?
+	if err.Error() != fmt.Sprintf("bucket '%s' not found", o.Config.Bucket.String) {
+		return err
+	}
+
+	// create a bucket with the default (infinite) retention policy
```

Comment:
the `infinite` part seems bad to me

Reply:
Would you add it as a configurable option?

Reply:
we used to have it as a configuration as well as the precision of the measurements and w/e "persistence" is in this context so we might want to actually read them and support them :shrug: